src/zarr_benchmarks/parse_json_for_plots.py

```diff
@@ -76,161 +79,334 @@ def get_benchmarks_dataframe(
 def plot_relplot_benchmarks(
     data: pd.DataFrame,
     *,
-    group: str,
     x_axis: str,
     y_axis: str,
-    hue: str,
-    size: str,
-    title: str,
-    output_filename: str | None = None,
+    sub_dir: str,
+    output_filename: str,
+    title: str | None = None,
+    hue: str | None = None,
+    size: str | None = None,
+    col: str | None = None,
 ) -> None:
+    if col is None:
+        facet_kws = None
+        col_wrap = None
+        plot_name = output_filename
+    else:
+        facet_kws = dict(sharex=True, sharey=True)
+        if len(data[col].unique()) < 3:
+            col_wrap = 2
+        else:
+            col_wrap = 3
+        plot_name = output_filename + "_subplots"
+
     graph = sns.relplot(
         data=data,
         x=x_axis,
         y=y_axis,
         hue=hue,
         style=hue,
         size=size,
+        col=col,
         height=4,
         aspect=1.5,
+        facet_kws=facet_kws,
+        col_wrap=col_wrap,
     )
-    x_axis_label, y_axis_label = get_axis_labels(x_axis, y_axis, group)
+    x_axis_label, y_axis_label = get_axis_labels(data, x_axis, y_axis)
     graph.set_axis_labels(x_axis_label, y_axis_label)
 
-    graph.figure.suptitle(title)
-    graph.figure.subplots_adjust(top=0.9)
-
-    if output_filename is not None:
-        save_plot_as_png(
-            graph,
-            Path(__file__).parents[2]
-            / "data"
-            / "plots"
-            / f"{group}_relplot_{output_filename}.png",
-        )
-
+    if title is not None:
+        graph.figure.suptitle(title)
+        graph.figure.subplots_adjust(top=0.9)
```

Comment:
Is it possible to use `graph.figure.tight_layout()` instead? tight_layout should be more robust to other changes in figure/axes sizes, and doesn't require 'guessing' the right parameters to set (in this case, currently `top=0.9`)